drools-wb-screens/drools-wb-dtable-xls-editor/drools-wb-dtable-xls-editor-backend/src/main/java/org/drools/workbench/screens/dtablexls/backend/server/DecisionTableXLSServiceImpl.java

```diff
@@ -108,256 +108,266 @@ public DecisionTableXLSServiceImpl( @Named("ioStrategy") final IOService ioServi
-                            comment,
-                            false );
+    public Path save(final Path resource,
+                     final InputStream content,
+                     final String sessionId,
+                     final String comment) {
+        return writeToFile(resource,
+                           content,
+                           sessionId,
+                           comment,
+                           false);
     }
 
-    private Path writeToFile( final Path resource,
-                              final InputStream content,
-                              final String sessionId,
-                              final String comment,
-                              boolean create ) {
-        final SessionInfo sessionInfo = getSessionInfo( sessionId );
+    private Path writeToFile(final Path resource,
+                             final InputStream content,
+                             final String sessionId,
+                             final String comment,
+                             boolean create) {
+        final SessionInfo sessionInfo = getSessionInfo(sessionId);
         String userAction = "UPDATING";
-        if ( create ) {
+        if (create) {
             userAction = "CREATING";
         }
-        log.info( "USER:" + sessionInfo.getIdentity().getIdentifier() + " " + userAction + " asset [" + resource.getFileName() + "]" );
+        log.info("USER:" + sessionInfo.getIdentity().getIdentifier() + " " + userAction + " asset [" + resource.getFileName() + "]");
 
         FileInputStream tempFIS = null;
         FileOutputStream tempFOS = null;
         OutputStream outputStream = null;
         try {
-            File tempFile = File.createTempFile( "testxls", null );
-            tempFOS = new FileOutputStream( tempFile );
-            IOUtils.copy( content, tempFOS );
+
+            File tempFile = File.createTempFile("testxls", null);
+            tempFOS = new FileOutputStream(tempFile);
+            IOUtils.copy(content, tempFOS);
             tempFOS.flush();
 
             //Validate the xls
-            validate( tempFile );
-
-            final org.uberfire.java.nio.file.Path nioPath = Paths.convert( resource );
-            if ( create ) {
-                ioService.createFile( nioPath );
+            validate(tempFile);
+
+            final org.uberfire.java.nio.file.Path nioPath = Paths.convert(resource);
+
+            ioService.startBatch(nioPath.getFileSystem());
+
+            if (create) {
+
+                tempFIS = new FileInputStream(tempFile);
+                ioService.write(nioPath,
+                                IOUtils.toByteArray(tempFIS),
+                                commentedOptionFactory.makeCommentedOption(comment,
+                                                                           sessionInfo.getIdentity(),
+                                                                           sessionInfo));
+            } else {
```

Comment:
I would expect some test double check `startBatch` is called. Same for `endBatch`. Also we just mock behavior of `ioService.write` without checking arguments. We capture `commentedOption` but not checking captured content.

All above makes me thing this is not tested appropriately.